javabuilder-authorizer/token_validator.rb

```diff
@@ -130,6 +125,10 @@ def teachers_over_hourly_limit?
         puts "teacher_associated_requests query has paginated responses. user_id #{@user_id} teacher_id #{teacher_id}"
       end
 
+      # Log the count of requests per hour to a metric. We are logging this for now so we can determine if we
+      # have set a good threshold for blocking classrooms.
+      @metrics_reporter.log_count_with_value(CLASSROOM_HOURLY_REQUEST_COUNT, response.count)
```

Comment:
should we log any other info here, like the teacher_id and/or user_id?

Reply:
This metric is meant for graphing in cloud watch and looking for general patterns across all users. It would be extra data we would probably never use to add user and teacher ids, and I'm not sure if it would mess with the graphs to have extra dimensions we always would want to aggregate over. If we want to dig into specific users I think we could query DynamoDB directly to try to find the problem classrooms

Reply:
Logging on every request seems like it might be a bit verbose. Would logging at a less frequent interval, say every time a classroom passes mod 1000 still give us the measure we're looking for?

Reply:
this is logging a metric which is meant to be verbose to my understanding. We could log less often but I thought it would be helpful to see the kind of usage we are getting, we can turn it off after a month or so. We already log multiple metrics on every run (performance times, lambda itself logs things like invocations counts).